jetty-core/jetty-http/src/main/java/org/eclipse/jetty/http/MimeTypes.java

```diff
@@ -876,4 +919,25 @@ else if (' ' != b)
             return value;
         return builder.toString();
     }
+
+    public static class ContentTypeField extends PreEncodedHttpField
```

Comment:
Wouldn't this make more sense as a standalone class?
After all we have all of the other specialized Fields as standalone classes, why is this one special?

Reply:
This `ContentTypeField` feels like the `HostPortHttpField`. (also a field used in HttpParser)

Reply:
I kind of agree that this would be better as standalone class, especially now that is the return value of a method: it's shorter to assign to a variable than `MimeTypes.ContentTypeField`.

Reply:
As this is a rather special purpose class, I've just made is package protected and provided static methods to extract the info if needed.     I could make it fully private if we want to give up on the optimized lookup of content-types with spaces in HttpParser